### DISCUSSION TITLE
Is there a way to call a method after ProjectToType<T> is completed?

### DISCUSSION THREAD

**Discussion by bhagavan44** [MapsterMapper/Mapster#698] [Q&A]
I wanted to modify a property value after the result is retrieved from the database with `ProjectToType` . `AfterMappingInline` and `AfterMapping` is not executed when we use the `ProjectToType`. What are the other options in this situation?

**Comment by savornicesei** [comment]
This is something I'm interested in. Have you found a solution, @bhagavan44 ?

**Reply by bhagavan44** [reply]
@savornicesei I was unable to find a suitable solution for that issue, so I decided to create my own code mapping tailored specifically to my use case.

**Reply by savornicesei** [reply]
@bhagavan44 Thank you. 

It is possible to use translations at mapping level but I doubt they'll work with `ProjectToType`
```
using Mapster;
using Microsoft.Extensions.Localization;
public class MappingConfig
{
    public static void ConfigureMappings(IStringLocalizer localizer)
    {
        TypeAdapterConfig<Entity, TranslatedDto>
            .NewConfig()
            .Map(dest => dest.TranslatedDescription, 
                 src => localizer[src.Description]); // Translate the Description field
    }
```
It is also possible to intercept materialized objects in EF Core and EF6 and apply translations there but that works only for table-entitites, not for composite ones.

**Comment by DocSvartz** [comment]
@bhagavan44  @savornicesei 
As far as I understand, ProjectToType<T> is needed to optimize the request, when the number of members in the destination type is less than in the source.
There is no point in using it to modify already received data.

It should work if localizer is a function `localizer(src.Description)`

>  ```
> TypeAdapterConfig<Entity, TranslatedDto>
>             .NewConfig()
>             .Map(dest => dest.TranslatedDescription, 
>                  src => localizer[src.Description]); // Translate the Description field
> ```
